CRM/Contribute/BAO/Contribution.php

```diff
@@ -2755,6 +2755,10 @@ public static function recordFinancialAccounts(&$params, $financialTrxnValues =
         $balanceTrxnParams['status_id'] = $statusId;
         $balanceTrxnParams['payment_instrument_id'] = $params['contribution']->payment_instrument_id;
         $balanceTrxnParams['check_number'] = CRM_Utils_Array::value('check_number', $params);
+        if (!empty($balanceTrxnParams['from_financial_account_id']) && 
+            ($statusId == array_search('Completed', $contributionStatuses) || $statusId == array_search('Partially paid', $contributionStatuses))) {
+          $balanceTrxnParams['is_payment'] = 1;
+        }
```

Comment:
Scenario :
Add a contribution from online as Credit card payment. Then the entry created in civicrm_financial_trxn will be
from_f_a = NULL
to_f_a = Asset account
payment_instrument= CC
Status = Completed
is_payment = 0.

Shouldn't the is_payment be 1?